### DISCUSSION TITLE
How do I use 2 credentials providers?

### DISCUSSION THREAD

**Discussion by Inzendis** [nextauthjs/next-auth#8937] [Help]
### Summary

So I saw the doc where it says you can use more than one. I want to use one where it takes email and password and another where it takes email, password, 2FA. 

Now I saw that you have you give it an Id and a a name to each, that I understand. But how do I tell it which one to use? I made my custom login page, and I use the signIn of import { signIn } from "next-auth/react"; there. It works as I want it, I just want to add that if the user has 2FA then enter the 2FA value, if not then just signIn with the regular one.

So my question is how do I tell it that I want the one with 2FA and not the other one?

### Additional information

I'm using Next,js 13 App Router

### Example
Where do I tell it do use the other one?
`  const res = await signIn("credentials", {
        redirect: false,
        email: formValues.email,
        password: formValues.password,
        callbackUrl,
      });`

**Comment by Inzendis** [comment]
if your id is "regularSignIn"

`const res = await signIn("regularSignIn", {
        redirect: false,
        email: formValues.email,
        password: formValues.password,
        callbackUrl,
      });`
      
      This will work, atleast it did for me.

**Reply by CHAIBNihal** [reply]
But if i have a schema for admin and other for client then i should to Work with 2 different credentialsProviders and of course I should to got different session next auth token but how i can do it
